jobrunner/local_run.py

```diff
@@ -335,30 +337,65 @@ def temporary_docker_auth_workaround(missing_docker_images):
     commands in any case, so this will hopefully be a short-lived hack.
     """
     if not os.environ.get("GITHUB_WORKFLOW"):
-        return missing_docker_images
-    stata_images = [
-        image
-        for image in missing_docker_images
-        if image.startswith("ghcr.io/opensafely/stata-mp:")
-    ]
-    if not stata_images:
-        return missing_docker_images
-    docker_config = os.environ.get("DOCKER_CONFIG", "~/.docker")
+        return False
+
+    docker_config = os.environ.get(
+        "DOCKER_CONFIG", os.path.expanduser("~/.docker")
+    )
     config_path = Path(docker_config) / "config.json"
     try:
         config = json.loads(config_path.read_text())
         auths = config["auths"]
     except Exception:
-        return missing_docker_images
+        return False
+
     if "ghcr.io" in auths or "docker.opensafely.org" not in auths:
-        return missing_docker_images
+        return False
+
     print("Applying Docker authentication workaround...")
-    for image in stata_images:
-        alt_image = image.replace("ghcr.io/opensafely/", "docker.opensafely.org/")
+    alt_image = image.replace("ghcr.io/opensafely/", "docker.opensafely.org/")
+
+    try:
         docker.pull(alt_image, quiet=True)
         print(f"Retagging '{alt_image}' as '{image}'")
         subprocess_run(["docker", "tag", alt_image, image])
-    return [image for image in missing_docker_images if image not in stata_images]
+    except Exception:
+        return False
+
+    return True
+
+
+def get_stata_license(repo=config.STATA_LICENSE_REPO):
+    """Load a stata license from local cache or remote repo."""
+    cached = Path(f'{tempfile.gettempdir()}/opensafely-stata.lic')
+    repos = [repo, repo.replace('https://', 'git+ssh://git@')]
+    env = {'GIT_TERMINAL_PROMPT': '0'} 
+
+    def git_clone(repo_url, cwd):
+        cmd = ['git', 'clone', '--depth=1', repo_url, 'repo']
+        result = subprocess_run(cmd, cwd=cwd, env={'GIT_TERMINAL_PROMPT': '0'})
```

Comment:
Do we want to capture output here? Otherwise don't we end up with git output in the terminal?